server/routes/handlers/getMatchingRecordRouteHandler.js

```diff
@@ -0,0 +1,67 @@
+const { settings } = require('../../config')
+const { getPaginationObject } = require('../helpers')
+const getMatchingRecordRouteHandler = (getMatchDetails, getCaseAndTemplateValues) => async (req, res) => {
+  const {
+    params: { courtCode, caseId, hearingId, defendantId },
+    session,
+    path,
+    query: queryParams
+  } = req
+
+  const templateValues = await getCaseAndTemplateValues(req)
+  templateValues.title = 'Review possible NDelius records'
+
+  const { data: { defendantName } } = templateValues
+
+  const response = await getMatchDetails(defendantId)
+
+  const { offenderMatchDetails } = response
+
+  if (!Array.isArray(offenderMatchDetails) || offenderMatchDetails.length === 0) {
```

Comment:
Is there guaranteed to be at least one match? We are throwing an error if there are no matches, is that correct?

Reply:
So the flow of screen is :

<img width="828" alt="Screenshot 2024-10-01 at 14 34 03" src="https://github.com/user-attachments/assets/02647b47-a031-4086-8f9f-a971a5061b80">


and then to 


<img width="913" alt="Screenshot 2024-10-01 at 14 34 20" src="https://github.com/user-attachments/assets/83694703-a7d4-464c-a675-0dabffd71c9b">


before coming to the below screen

<img width="1479" alt="Screenshot 2024-10-01 at 15 06 35" src="https://github.com/user-attachments/assets/d68fc9b5-2f12-4cf1-ac0f-f8af3d4319dc">

My assumption is that, by the time we are coming to this screen we have at least one match.